### DISCUSSION TITLE
Customise column type parameters for reflection

### DISCUSSION THREAD

**Discussion by ghost** [sqlalchemy/sqlalchemy#6141] [Usage Questions] [answered]
**Is your feature request related to a problem? Please describe.**
Example: Reflection of a Postgres DB that uses `UUID` will result in `Column` with `UUID(as_uuid=False)` as the type, which causes SQLAlchemy to call `.replace` on my UUID objects when I insert.

**Describe the solution you'd like**
Some hook I can provide to `automap_base`? I'm not sure what would be the best.

**Describe alternatives you've considered**
I'm aware I may use `extend_existing=True`, but this would be a pain if the columns change regularly or you need to connect to multiple different PG databases. The sense I get from reading the docs is that  providing Table objects is for when the DB is missing something you know your code should follow. PG is not missing anything in this case, so it defeats the point of the reflection process, which is to work with what the DB can tell it.

Bonus question: Will anything break if I replace type of the column after the Table is made? Because then I could just iterate through all the Table objects's columns and swap them out for now.

Second bonus question: Why can't we call `isinstance(v, UUID)` and avoid `as_uuid`  entirely?

**Comment by zzzeek** [comment, marked as answer]
> **Is your feature request related to a problem? Please describe.**
> Example: Reflection of a Postgres DB that uses `UUID` will result in `Column` with `UUID(as_uuid=False)` as the type, which causes SQLAlchemy to call `.replace` on my UUID objects when I insert.
> 
> **Describe the solution you'd like**
> Some hook I can provide to `automap_base`? I'm not sure what would be the best.

The automap base documents this hook, which is not specific to automap, at https://docs.sqlalchemy.org/en/14/orm/extensions/automap.html#intercepting-column-definitions .   Inspect the "type" member of the dictionary for the UUID type and set as_uuid=True.

> 
> Bonus question: Will anything break if I replace type of the column after the Table is made? Because then I could just iterate through all the Table objects's columns and swap them out for now.

probably not but as above this is unnecessary.

> 
> Second bonus question: Why can't we call `isinstance(v, UUID)` and avoid `as_uuid` entirely?

This would add 50000 function calls to a SQL statement that returns 10000 rows with 5 UUID values in each row.
